src/kvstore/raftex/RaftPart.h

```diff
@@ -325,6 +335,14 @@ class RaftPart : public std::enable_shared_from_this<RaftPart> {
             return singlePromises_.back().getFuture();
         }
 
+        folly::Future<ValueType> getAndRollSharedFuture() {
```

Comment:
Not sure why you need to use SharedPromise here. I think what you need is a SinglePromise, just like CAS does